src/hotspot/share/runtime/objectMonitor.inline.hpp

```diff
@@ -92,22 +92,22 @@ inline void ObjectMonitor::clear_common() {
   assert(contentions() <= 0, "must not be positive: contentions=%d", contentions());
   assert(_waiters == 0, "must be 0: waiters=%d", _waiters);
   assert(_recursions == 0, "must be 0: recursions=" INTX_FORMAT, _recursions);
-  assert(_object != NULL, "must be non-NULL");
+  assert(object() != NULL, "must be non-NULL");
```

Comment:
In your next changeset, you'll want object to peek not resolve when it's an OopHandle.

Reply:
Actually that assert is gone in JDK-8247281. Since the _object field becomes
a weak handle, we cannot assert that object() is not NULL there.